realm-annotations-processor/src/main/java/io/realm/processor/RealmJsonTypeHelper.java

```diff
@@ -37,32 +36,41 @@
         JAVA_TO_JSON_TYPES.put("float", new SimpleTypeConverter("float", "Double"));
         JAVA_TO_JSON_TYPES.put("double", new SimpleTypeConverter("double", "Double"));
         JAVA_TO_JSON_TYPES.put("boolean", new SimpleTypeConverter("boolean", "Boolean"));
-        JAVA_TO_JSON_TYPES.put("Byte", new SimpleTypeConverter("Byte", "Int"));
-        JAVA_TO_JSON_TYPES.put("Short", new SimpleTypeConverter("Short", "Int"));
-        JAVA_TO_JSON_TYPES.put("Integer", new SimpleTypeConverter("Integer", "Int"));
-        JAVA_TO_JSON_TYPES.put("Long", new SimpleTypeConverter("Long", "Long"));
-        JAVA_TO_JSON_TYPES.put("Float", new SimpleTypeConverter("Float", "Double"));
-        JAVA_TO_JSON_TYPES.put("Double", new SimpleTypeConverter("Double", "Double"));
-        JAVA_TO_JSON_TYPES.put("Boolean", new SimpleTypeConverter("Boolean", "Boolean"));
+        JAVA_TO_JSON_TYPES.put("java.lang.Byte", new SimpleTypeConverter("byte", "Int"));
+        JAVA_TO_JSON_TYPES.put("java.lang.Short", new SimpleTypeConverter("short", "Int"));
+        JAVA_TO_JSON_TYPES.put("java.lang.Integer", new SimpleTypeConverter("int", "Int"));
+        JAVA_TO_JSON_TYPES.put("java.lang.Long", new SimpleTypeConverter("long", "Long"));
+        JAVA_TO_JSON_TYPES.put("java.lang.Float", new SimpleTypeConverter("float", "Double"));
+        JAVA_TO_JSON_TYPES.put("java.lang.Double", new SimpleTypeConverter("double", "Double"));
+        JAVA_TO_JSON_TYPES.put("java.lang.Boolean", new SimpleTypeConverter("boolean", "Boolean"));
         JAVA_TO_JSON_TYPES.put("java.lang.String", new SimpleTypeConverter("String", "String"));
         JAVA_TO_JSON_TYPES.put("java.util.Date", new JsonToRealmTypeConverter() {
             @Override
-            public void emitTypeConversion(String setter, String fieldName, String fieldType, JavaWriter writer) throws IOException {
+            public void emitTypeConversion(String setter, String fieldName, String fieldType, JavaWriter writer,
+                                           boolean nullable) throws IOException {
```

Comment:
Thanks for the catch! `nullable` arg is not needed anymore. It was there because of at that time I was trying to throw exception for the required boxed type. Removed in the coming commit.